recipes/open62541/all/conanfile.py

```diff
@@ -138,47 +140,50 @@ class Open62541Conan(ConanFile):
 
     exports_sources = ["CMakeLists.txt", "patches/**"]
     exports = "submoduledata.yml"
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
 
     short_paths = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if tools.Version(self.version) >= "1.3.1":
+
+        if Version(self.version) >= "1.3.1":
             del self.options.embedded_profile
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
         if not self.options.cpp_compatible:
-            del self.settings.compiler.cppstd
-            del self.settings.compiler.libcxx
+            self.settings.rm_safe("compiler.libcxx")
+            self.settings.rm_safe("compiler.cppstd")
 
         # Due to https://github.com/open62541/open62541/issues/4687 we cannot build with 1.2.2 + Windows + shared
-        if tools.Version(self.version) >= "1.2.2" and self.settings.os == "Windows" and self.options.shared:
+        if Version(self.version) >= "1.2.2" and self.settings.os == "Windows" and self.options.shared:
             raise ConanInvalidConfiguration("{0} {1} doesn't properly support shared lib on Windows".format(self.name,
                                                                                                             self.version))
 
         if self.options.subscription == "With Events":
-            self.output.warning("`{name}:subscription=With Events` is deprecated. Use `{name}:subscription=events` instead".format(name=self.name))  # Deprecated in 1.2.2
+            self.output.warning("`{name}:subscription=With Events` is deprecated. Use `{name}:subscription=events` instead".format(
+                name=self.name))  # Deprecated in 1.2.2
             self.options.subscription = "events"
 
         if self.options.web_socket:
             self.options["libwebsockets"].with_ssl = self.options.encryption
 
+    def layout(self):
+        cmake_layout(self, src_folder="source")
+
     def requirements(self):
         if self.options.encryption == "mbedtls":
-            self.requires("mbedtls/2.25.0")
+            self.requires("mbedtls/3.2.1")
         elif self.options.encryption == "openssl":
```

Comment:
I wouldn't change majors 🙊 unless the project supports it (please add a link if ot does)

Reply:
@prince-chrismc, sadly to my knowledge, open62541 community does not specify which **mbedtls** or **openssl** dependency version. 

I suspect that it works with the latest available version, but I will test this locally for the earliest version as well as the current one in the PR

Reply:
hey @prince-chrismc, you were right on the major versions, it seems that open62541 does not like either openssl 3.* or mbedtls 3.* releases, I'll use the latest available minor releases for opnessl 1.* and mbedtls 2.*

Reply:
👍